[Isaac Tan] iP

iztanpy opened the pull request:
# DUKEPRO
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/nus-cs2103-AY2122S2/ip/pull/41).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

iztanpy committed:
Level-1

iztanpy committed:
Level-2

iztanpy committed:
Level-3

iztanpy committed:
Level-4

iztanpy committed:
A-TextUiTesting

iztanpy committed:
level-5

iztanpy committed:
Level-6

iztanpy committed:
A-Enums

iztanpy committed:
Level-7

iztanpy committed:
Level-7

iztanpy committed:
Level-8

iztanpy committed:
Merge branch 'branch-Level-7'

iztanpy committed:
Update Merging Errors

iztanpy committed:
A-MoreOOP

iztanpy committed:
A-MoreOOP

iztanpy committed:
A-Packages

iztanpy committed:
update post packages

iztanpy committed:
A-Packages

iztanpy committed:
A-JUnit

iztanpy committed:
A-Jar

iztanpy committed:
A-Jar

iztanpy committed:
added A-javaDoc

iztanpy committed:
A-CodingStandard

iztanpy committed:
Level-9

iztanpy committed:
Merge branch 'branch-A-codingStandard'

iztanpy committed:
Merge branch 'branch-A-javaDocs'

iztanpy committed:
Update list counter to fix error with indexing

nnmoq commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,40 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import tasks.*;
```
```suggestion
import tasks.Deadline;
import tasks.Event;
import tasks.Task;
import tasks.Todo;
```

nnmoq commented on src/main/java/backend/FileDecoder.java:
```diff
@@ -0,0 +1,65 @@
+        // if String supplied is one with a date
+        if (args.length == 4) {
+            try {
+                SimpleDateFormat dateFormatter = new SimpleDateFormat("MMM dd yyyy");
```
I suggest you bring 
```
                SimpleDateFormat dateFormatter = new SimpleDateFormat("MMM dd yyyy");
```
out as a global constant for Task class so you don't have to create a new SimpleDateFormat everytime decode(String tasks) is called.
This is just for efficiency though.

nnmoq commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,66 @@
+import java.util.Scanner;
+
+public class Storage {
+    private final File save;
```
```suggestion
    private final File SAVE_FILE;
```

nnmoq commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,66 @@
+
+public class Storage {
+    private final File save;
+    private final ArrayList<Task> tasks = new ArrayList<>();
```
```suggestion
    private final ArrayList<Task> TASKS = new ArrayList<>();
```

nnmoq commented on src/main/java/backend/Commands.java:
```diff
@@ -0,0 +1,5 @@
+package backend;
+
+public enum Commands {
```
Nice use of enums. 👍

nnmoq reviewed (commented):
Seems pretty good, nice use of oop and enums.
One suggestion is that you need to specify each import as we can't use .* even if you are using all the imports in that package.

YiFei2 commented on src/main/java/backend/InputDecoder.java:
```diff
@@ -0,0 +1,75 @@
+        String[] commandSections = input.split(" ", 2);
+        currentCommand = Commands.valueOf(commandSections[0].toUpperCase());
+        return commandSections;
+    }
```
I like the abstraction of parsing input here!

YiFei2 commented on src/main/java/backend/TaskList.java:
```diff
@@ -0,0 +1,159 @@
+
+    /**
+     * Marks the task at the supplied integer as completed, based on their ordering in this.list().
+     * @param indexMarked index of the completed task
```
I think there should be an empty line between description and the parameter section here

YiFei2 commented on src/main/java/backend/TaskList.java:
```diff
@@ -0,0 +1,159 @@
+
+    /**
+     * Unmarks the task at the supplied integer as completed, based on their ordering in this.list().
+     * @param indexUnmarked index of the task to be unmarked
```
```suggestion
     * Unmarks the task at the supplied integer as completed, based on their ordering in this.list().
     *
     * @param indexUnmarked index of the task to be unmarked
```

YiFei2 commented on src/main/java/backend/TaskList.java:
```diff
@@ -0,0 +1,159 @@
+        for (int i = 0; i < tasks.size(); i++) {
+            Task currentTask = tasks.get(i);
+            int index = i + 1;
+            System.out.println(index + "." + currentTask.toString());
```
Perhaps this could be added into your Ui.list() function?

YiFei2 reviewed (commented):
Looks good!

iztanpy committed:
re-implemented junit testing

iztanpy committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

iztanpy committed:
A-CheckStyle

iztanpy committed:
Added GUI to Duke

iztanpy committed:
Fixed minor bugs for GUI

iztanpy committed:
applied some padding to the GUI

iztanpy committed:
InputDecoder decode:  No checks that user inputs an integer for commands that require an index (mark, unmark, delete)

Currently, no prompt is given to the user when the user tries a command like "delete all", which may not be new user friendly

Let's,
* Provide a prompt to the user that reminds them to use mark/unmark/delete in conjunction with the task's respective index

iztanpy committed:
General: Currently no assertions

Unable to check if certain parts of the application may break if updates are done to methods from different classes

Let's:
* Add Assertions into the delete/mark/unmark commands, to check if they are working accurately
* Add Assertions into the list command, to check if all existing tasks are being listed

iztanpy committed:
General: Code quality not up to code standards,

some of the javadocs have become out of date after editing existing code base

Lets:
* Update javadocs description to better match current function output
* Fix javadocs formatting, add a blank line between params and description
* change some variables names to make them better in line with coding standard

iztanpy committed:
Merge pull request #1 from iztanpy/branch-A-CodeQuality

General: Code quality not up to code standards

iztanpy committed:
Merge branch 'master' of https://github.com/iztanpy/ip into branch-A-Assertions

* 'master' of https://github.com/iztanpy/ip:
  General: Code quality not up to code standards,

iztanpy committed:
BCD-Extension

iztanpy committed:
A-BetterGui

iztanpy committed:
added ui and readme

iztanpy committed:
Set theme jekyll-theme-minimal
